### ISSUE TITLE
Migrating from Drupal 10.1 to 10.2 fails with PDOException: SQLSTATE[42S02]: Base table or view not found

### ISSUE THREAD

**Issue by fotiDim** [omega8cc/boa#1776]
Migrating from Drupal  10.1 to 10.2 fails with:
```
PDOException: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'testdomain_0.batch' doesn't exist in /data/disk/o1/static/no-modules-b/web/core/lib/Drupal/Core/Database/StatementWrapperIterator.php:111
Stack trace:
#0 /data/disk/o1/static/no-modules-b/web/core/lib/Drupal/Core/Database/StatementWrapperIterator.php(111): PDOStatement->execute()
#1 /data/disk/o1/static/no-modules-b/web/core/lib/Drupal/Core/Database/Query/Update.php(152): Drupal\Core\Database\StatementWrapperIterator->execute()
#2 /data/disk/o1/static/no-modules-b/web/core/lib/Drupal/Core/Batch/BatchStorage.php(146): Drupal\Core\Database\Query\Update->execute()
#3 /data/disk/o1/static/no-modules-b/web/core/lib/Drupal/Core/ProxyClass/Batch/BatchStorage.php(107): Drupal\Core\Batch\BatchStorage->create()
#4 /data/disk/o1/tools/drush/commands/core/drupal/batch_10.inc(52): Drupal\Core\ProxyClass\Batch\BatchStorage->create()
#5 /data/disk/o1/tools/drush/includes/batch.inc(81): _drush_backend_batch_process()
#6 /data/disk/o1/tools/drush/commands/core/drupal/update_10.inc(317): drush_backend_batch_process()
#7 /data/disk/o1/tools/drush/commands/core/drupal/update_10.inc(202): drush_update_batch()
#8 /data/disk/o1/tools/drush/commands/core/core.drush.inc(466): update_main()
#9 /data/disk/o1/tools/drush/includes/command.inc(422): drush_core_updatedb()
#10 /data/disk/o1/tools/drush/includes/command.inc(231): _drush_invoke_hooks()
#11 /data/disk/o1/tools/drush/includes/command.inc(199): drush_command()
#12 /data/disk/o1/tools/drush/lib/Drush/Boot/BaseBoot.php(67): drush_dispatch()
#13 /data/disk/o1/tools/drush/includes/preflight.inc(67): Drush\Boot\BaseBoot->bootstrap_and_dispatch()
#14 /data/disk/o1/tools/drush/drush.php(12): drush_main()
#15 {main}

Next Drupal\Core\Database\DatabaseExceptionWrapper: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'testdomain_0.batch' doesn't exist: UPDATE "batch" SET "token"=:db_update_placeholder_0, "batch"=:db_update_placeholder_1
WHERE "bid" = :db_condition_placeholder_0; Array
(
    [:db_update_placeholder_0] => T7oVDQT8jI-NF8d-7P3Cm2JB67M6GatZud08NZGmqFk
    [:db_update_placeholder_1] => a:5:{s:4:"sets";a:1:{i:0;a:14:{s:7:"sandbox";a:0:{}s:7:"results";a:0:{}s:7:"success";b:0;s:5:"start";i:0;s:7:"elapsed";i:0;s:5:"title";s:8:"Updating";s:12:"init_message";s:27:"Starting updates ";s:13:"error_message";s:131:"An unrecoverable error has occurred. You can find the error message below. It is advised to copy it to the clipboard for reference.";s:8:"finished";s:21:"drush_update_finished";s:4:"file";s:19:"includes/update.inc";s:16:"progress_message";O:48:"Drupal\Core\StringTranslation\TranslatableMarkup":3:{s:9:"*string";s:29:"Completed @current of @total.";s:12:"*arguments";a:0:{}s:10:"*options";a:0:{}}s:5:"total";i:25;s:5:"count";i:25;s:5:"queue";a:2:{s:4:"name";s:16:"drupal_batch:2:0";s:5:"class";s:23:"Drupal\Core\Queue\Batch";}}}s:16:"has_form_submits";b:0;s:11:"current_set";i:0;s:2:"id";s:1:"2";s:11:"progressive";b:1;}
    [:db_condition_placeholder_0] => 2
)
 in /data/disk/o1/static/no-modules-b/web/core/modules/mysql/src/Driver/Database/mysql/ExceptionHandler.php:56
Stack trace:
#0 /data/disk/o1/static/no-modules-b/web/core/lib/Drupal/Core/Database/Query/Update.php(156): Drupal\mysql\Driver\Database\mysql\ExceptionHandler->handleExecutionException()
#1 /data/disk/o1/static/no-modules-b/web/core/lib/Drupal/Core/Batch/BatchStorage.php(146): Drupal\Core\Database\Query\Update->execute()
#2 /data/disk/o1/static/no-modules-b/web/core/lib/Drupal/Core/ProxyClass/Batch/BatchStorage.php(107): Drupal\Core\Batch\BatchStorage->create()
#3 /data/disk/o1/tools/drush/commands/core/drupal/batch_10.inc(52): Drupal\Core\ProxyClass\Batch\BatchStorage->create()
#4 /data/disk/o1/tools/drush/includes/batch.inc(81): _drush_backend_batch_process()
#5 /data/disk/o1/tools/drush/commands/core/drupal/update_10.inc(317): drush_backend_batch_process()
#6 /data/disk/o1/tools/drush/commands/core/drupal/update_10.inc(202): drush_update_batch()
#7 /data/disk/o1/tools/drush/commands/core/core.drush.inc(466): update_main()
#8 /data/disk/o1/tools/drush/includes/command.inc(422): drush_core_updatedb()
#9 /data/disk/o1/tools/drush/includes/command.inc(231): _drush_invoke_hooks()
#10 /data/disk/o1/tools/drush/includes/command.inc(199): drush_command()
#11 /data/disk/o1/tools/drush/lib/Drush/Boot/BaseBoot.php(67): drush_dispatch()
#12 /data/disk/o1/tools/drush/includes/preflight.inc(67): Drush\Boot\BaseBoot->bootstrap_and_dispatch()
#13 /data/disk/o1/tools/drush/drush.php(12): drush_main()
#14 {main}
```

Creating a new site on 10.2 works fine. It's only migration that fails.

Steps to reproduce.
- Create 10.1 platform with `composer create-project drupal/recommended-project 10dot1`, then change those in composer.json:
```
        "drupal/core-composer-scaffold": "10.1.8",
        "drupal/core-project-message": "10.1.8",
        "drupal/core-recommended": "10.1.8"
```
run `composer update`,
Use `Deploy a Composer project from a Git repository` when creating the platform or `Deploy from Git repository (not managed by Composer)`.
- Create 10.2 platform with `composer create-project drupal/recommended-project 10dot2`. It will default to 10.2.4. Use `Deploy a Composer project from a Git repository` or `Deploy from Git repository (not managed by Composer)`. Same as above.
- Create site with 10.1 platform using the Standard profile
- Try to migrate site to 10.2

**Comment by omega8cc** [comment]
None of this happens in DEV. New release is coming.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by omega8cc** [comment]
This has been fixed in the 5.2.0 release.
